context/FarmContext/helpers.ts

```diff
@@ -2,23 +2,19 @@ import { initPool, fetchCommits } from '@context/PoolContext/helpers';
 import { calcNextValueTransfer, calcTokenPrice } from '@tracer-protocol/tracer-pools-utils';
 import { ethers } from 'ethers';
 import BigNumber from 'bignumber.js';
+import { StaticPoolInfo } from '@libs/types/General';
+import { DEFAULT_POOLSTATE } from '@libs/constants/pool';
 
 export const fetchTokenPrice: (
-    pool: string,
+    poolInfo: StaticPoolInfo,
     tokenAddresses: [string] | [string, string],
     provider: ethers.providers.JsonRpcProvider | undefined,
-) => Promise<BigNumber[]> = async (pool, tokenAddresses, provider) => {
-    if (!provider) {
+) => Promise<BigNumber[]> = async (poolInfo_, tokenAddresses, provider) => {
+    if (!provider || !poolInfo_) {
         return [new BigNumber(1)];
     }
 
-    const poolInfo = await initPool(
-        {
-            name: '',
-            address: pool,
-        },
-        provider,
-    );
+    const poolInfo = await initPool(poolInfo_, provider);
```

Comment:
just noticed I don't think there is any attempt to cache results between calls to `initPools` at all. Don't think it will be an issue though because it looks like fetchTokenPrice only gets called during initilization of the farms